pdns/rec-lua-conf.cc

```diff
@@ -316,42 +353,19 @@ void loadRecursorLuaConfig(const std::string& fname, bool checkOnly)
     });
 
 #if HAVE_PROTOBUF
-  Lua.writeFunction("protobufServer", [&lci, checkOnly](const string& server_, const boost::optional<uint16_t> timeout, const boost::optional<uint64_t> maxQueuedEntries, const boost::optional<uint8_t> reconnectWaitTime, const boost::optional<uint8_t> maskV4, boost::optional<uint8_t> maskV6, boost::optional<bool> asyncConnect, boost::optional<bool> taggedOnly) {
+  Lua.writeFunction("setProtobufMasks", [&lci](const uint8_t maskV4, uint8_t maskV6) {
+      lci.protobufMaskV4 = maskV4;
+      lci.protobufMaskV6 = maskV6;
+    });
+
+  Lua.writeFunction("protobufServer", [&lci](const string& server_, boost::optional<protobufOptions_t> vars) {
       try {
-	ComboAddress server(server_);
+        lci.protobufExportConfig.server = ComboAddress(server_);
```

Comment:
This will change protobuf server's IP address no matter it has already been enabled or not. I think it should be moved after line 365 (`if (!lci.protobufExportConfig.enabled) {`).

Reply:
Nice catch, thanks!